src/java.base/macosx/classes/sun/nio/fs/BsdFileSystem.java

```diff
@@ -54,8 +52,7 @@ class BsdFileSystem extends UnixFileSystem {
     public WatchService newWatchService()
         throws IOException
     {
-        // use polling implementation until we implement a BSD/kqueue one
-        return new PollingWatchService();
+        return new MacOSXWatchService();
```

Comment:
Is it still intended that there be a system property to enable `MacOSXWatchService` which would be off by default?

Reply:
There was such a suggestion (not sure about "off by default", though). I guess this should go through CSR review, in which case I'm going to need some guidance.

Reply:
If you file a CSR then I can help with editing.

Reply:
Filed https://bugs.openjdk.org/browse/JDK-8296164
@bplb Could you please take a look and edit parts that need it?

Reply:
@mkartashev I made some minor changes to the CSR. Was the property name previously agreed upon? I would tend to drop the `.use` from the end of it.

Reply:
> Was the property name previously agreed upon?

Nope. Please, rename as you see fit, I really have no preference nor do I have any experience in CSRs.

Reply:
If we can get the new implementation stable then it might not be worth including the polling implementation in the macOS build. The make file that controls whether PollingWatchService is included or not is in make/modules/java.base/Java.gmk.